src/main/java/com/onlinecommunity/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,53 @@
+package com.onlinecommunity.security;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ObjectUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+
+import java.io.IOException;
+
+// 입력받은 토큰을 기반으로 유효성 검사하는 기능을 하는 클래스
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    public static final String TOKEN_HEADER = "Authorization";
+    public static final String TOKEN_PREFIX = "Bearer ";
+
+    private final TokenProvider tokenProvider;
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        String token = this.resolveTokenFromRequest(request);
+
+        if(StringUtils.hasText(token) && this.tokenProvider.validateToken(token)) {
```

Comment:
```suggestion
        if (StringUtils.hasText(token) && this.tokenProvider.validateToken(token)) {
```
if 문에 대한 coding convention 은 위와 같으니 지켜보시면 좋겠네요~!
https://naver.github.io/hackday-conventions-java/